src/pyhf/infer/mle.py

```diff
@@ -47,6 +47,17 @@ def twice_nll(pars, data, pdf):
     return -2 * pdf.logpdf(pars, data)
 
 
+def _validate_fit_inputs(init_pars, par_bounds, fixed_params):
+    for par_idx, (value, bound) in enumerate(zip(init_pars, par_bounds)):
+        if not (bound[0] <= value <= bound[1]):
+            raise ValueError(
+                f"fit initialization parameter (index: {par_idx}, value: {value}) lies outside of its bounds: {bound}"
+                + "\nTo correct this adjust the initialization parameter values in the model spec or those given"
+                + "\nas arguments to pyhf.infer.fit. If this value is intended, adjust the range of the parameter"
+                + "\nbounds."
```

Comment:
I made the error message more verbose, and hopefully more informative. If you run the example `issue_1123.py` in Issue #1123 you now get

```
---
bounds on POI: [0, 5]
Optimizer: MINUIT
Traceback (most recent call last):
  File "issue_1123.py", line 64, in <module>
    main([0, 5], use_minuit=True)
  File "issue_1123.py", line 57, in main
    f"test_mu: {test_mu}, CLs_obs: {pyhf.infer.hypotest(test_mu, data, model, qtilde=True)}"
  File "/home/feickert/Code/GitHub/pyhf/src/pyhf/infer/__init__.py", line 133, in hypotest
    teststat = calc.teststatistic(poi_test)
  File "/home/feickert/Code/GitHub/pyhf/src/pyhf/infer/calculators.py", line 195, in teststatistic
    self.fixed_params,
  File "/home/feickert/Code/GitHub/pyhf/src/pyhf/infer/test_statistics.py", line 177, in qmu_tilde
    return _qmu_like(mu, data, pdf, init_pars, par_bounds, fixed_params)
  File "/home/feickert/Code/GitHub/pyhf/src/pyhf/infer/test_statistics.py", line 20, in _qmu_like
    mu, data, pdf, init_pars, par_bounds, fixed_params, return_fitted_pars=True
  File "/home/feickert/Code/GitHub/pyhf/src/pyhf/infer/test_statistics.py", line 39, in _tmu_like
    mu, data, pdf, init_pars, par_bounds, fixed_params, return_fitted_val=True
  File "/home/feickert/Code/GitHub/pyhf/src/pyhf/infer/mle.py", line 190, in fixed_poi_fit
    return fit(data, pdf, init_pars, par_bounds, fixed_params, **kwargs)
  File "/home/feickert/Code/GitHub/pyhf/src/pyhf/infer/mle.py", line 113, in fit
    _validate_fit_inputs(init_pars, par_bounds, fixed_params)
  File "/home/feickert/Code/GitHub/pyhf/src/pyhf/infer/mle.py", line 57, in _validate_fit_inputs
    + "\nbounds."
ValueError: fit initialization parameter (index: 0, value: 6.0) lies outside of its bounds: [0, 5]
To correct this adjust the initialization parameter values in the model spec or those given
as arguments to pyhf.infer.fit. If this value is intended, adjust the range of the parameter
bounds.
```